packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -489,13 +574,150 @@ class SliderThemeData extends Diagnosticable {
+    properties.add(DiagnosticsProperty<SliderComponentShape>('overlayShape', overlayShape, defaultValue: defaultData.overlayShape, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<SliderComponentShape>('valueIndicatorShape', valueIndicatorShape, defaultValue: defaultData.valueIndicatorShape, level: DiagnosticLevel.debug));
     properties.add(EnumProperty<ShowValueIndicator>('showValueIndicator', showValueIndicator, defaultValue: defaultData.showValueIndicator));
     properties.add(DiagnosticsProperty<TextStyle>('valueIndicatorTextStyle', valueIndicatorTextStyle, defaultValue: defaultData.valueIndicatorTextStyle));
   }
 }
 
+/// Base class for slider track shapes.
+///
+/// Track shapes define the appearance of the slider track.
+///
+/// The [getPreferredRect] method is used to provide horizontal boundaries for
+/// the thumb's position, and to help position the thumb and ticks on the track.
+///
+/// See also:
+///
+///  * [RectangularSliderTrackShape] for the default track shape.
+///  * [SliderTickShape] for the base class for custom a tick shape.
+///  * [SliderComponentShape] for the base class for custom a component shape.
+abstract class SliderTrackShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const SliderTrackShape();
+
+  /// Returns the preferred bounds of the shape.
+  ///
+  /// It is used to provide horizontal boundaries for the thumb's position,
+  /// and to help position the thumb and ticks on the track.
+  ///
+  /// [parentBox] is the box of the [_RenderSlider] itself.
+  ///
+  /// [offset] is the offset relative to this shape's [PaintingContext].
+  ///
+  /// [sliderTheme] is the theme assigned to the [Slider] that this track shape
+  /// belongs to.
+  ///
+  /// [isEnabled] is whether or not the the [Slider] is enabled.
+  ///
+  /// [isDiscrete] is whether or not the the [Slider] is discrete.
+  Rect getPreferredRect({
+    RenderBox parentBox,
+    Offset offset = Offset.zero,
+    SliderThemeData sliderTheme,
+    bool isDiscrete,
+  });
+
+  /// Paints the track shape based on the state passed to it.
+  ///
+  /// [offset] is the offset relative to this shape's [PaintingContext].
+  ///
+  /// [parentBox] is the box of the [_RenderSlider] itself.
+  ///
+  /// [sliderTheme] is the theme assigned to the [Slider] that this track shape
+  ///  belongs to.
+  ///
+  /// [enableAnimation] is an animation triggered when the [Slider] is enabled,
+  /// and it reverses when the slider is disabled. Enabled is the
+  /// [Slider.isInteractive] state. Use this to paint intermediate track frames
+  /// when the slider changes enabled state.
+  ///
+  /// [trackDividers] are the values between 0.0 and 1.0 where the track is
+  /// divided by any thumbs.
+  ///
+  /// [trackSegmentsActive] are the bools that represent if a track segment
```

Comment:
Since the SliderTrackShape is new, there is a chance to make the API future backwards-compatible for more custom sliders.

If the only valid active track patterns will always be limited to [true, false] and [false, true, false], then this param is not needed. 

However, this API is currently a bit more open, without causing any additional complexity or expectations of the developer that is using it. For example, if the 2 thumb slider wanted to be an inverted range slider, then the callee of this shape's paint method would simply pass in [true, false, true] for this value. This is only needed the RangeSlider (or RawSlider, if there is a need for it) exposes this param in its constructor, which is something that is still open in the design.

Reply:
Updated based on conversation we had. It is now more like the value indicator by using text direction to determine the active pieces, and only the 1 thumb location is used for 2 track segments